pkg/ccl/backupccl/split_and_scatter_processor.go

```diff
@@ -82,22 +92,26 @@ func (s dbSplitAndScatterer) split(
 	if s.kr == nil {
 		return errors.AssertionFailedf("KeyRewriter was not set when expected to be")
 	}
-	if s.db == nil {
+	if s.sas == nil {
 		return errors.AssertionFailedf("split and scatterer's database was not set when expected")
 	}
 
-	expirationTime := s.db.Clock().Now().Add(time.Hour.Nanoseconds(), 0)
+	expirationTime := s.sas.Clock().Now().Add(time.Hour.Nanoseconds(), 0)
 	newSplitKey, err := rewriteBackupSpanKey(codec, s.kr, splitKey)
 	if err != nil {
 		return err
 	}
 	if splitAt, err := keys.EnsureSafeSplitKey(newSplitKey); err != nil {
-		// Ignore the error, not all keys are table keys.
+		// The key might be corrupt, and therefore we cannot guarantee it is a valid
+		// split key. The restore can still continue without this split. This error
+		// is not expected after #109483 is fixed.
+		log.Errorf(ctx, "failed splitting at key %s err: %v", newSplitKey, err)
+		return nil
 	} else if len(splitAt) != 0 {
 		newSplitKey = splitAt
 	}
-	log.VEventf(ctx, 1, "presplitting new key %+v", newSplitKey)
-	if err := s.db.AdminSplit(ctx, newSplitKey, expirationTime); err != nil {
+	log.Infof(ctx, "presplitting on safe key %s", newSplitKey)
```

Comment:
did you intend to switch this to info? also, I'm not sure if "safe" if meaningful to reader of the logs here, so I might just revert this line.

Reply:
> did you intend to switch this to info? 

yes, I was thinking: 10K splits on a node, 200ish bytes per message, gives us 2MB of spam, not great but really nice to see the actual split that restore asked for. Would make debugging the next unsafe split issue easier. WDYT?

> I'm not sure if "safe" if meaningful to reader of the logs here

yep, good point, this was "new key" which I also thought doesn't mean much? changed now to just "key".

Reply:
I'm a little wary of adding net new behavior to a backport, including 10k log lines during a restore; seems like more than is strictly required to fix the bug.

Reply:
ah yes, you did say that, oops.
done. thanks!